verilog/CST/verilog_tree_print.h

```diff
@@ -41,6 +44,29 @@ class VerilogPrettyPrinter : public verible::PrettyPrinter {
 void PrettyPrintVerilogTree(const verible::Symbol& root, absl::string_view base,
                             std::ostream* stream);
 
+class VerilogTreeToJsonConverter : public verible::SymbolVisitor {
+ public:
+  explicit VerilogTreeToJsonConverter(absl::string_view base);
+
+  void Visit(const verible::SyntaxTreeLeaf&) override;
+  void Visit(const verible::SyntaxTreeNode&) override;
+
+  Json::Value get_json() { return json_; }
+
+ protected:
+  // Range of text spanned by syntax tree, used for offset calculation.
+  const verible::TokenInfo::Context context_;
+
+  // JSON tree root
+  Json::Value json_;
+
+  // Pointer to JSON value of currently visited symbol in its parent's
+  // children list.
+  Json::Value* value_;
+};
+
+Json::Value ConvertVerilogTreeToJson(const verible::Symbol& root, absl::string_view base);
```

Comment:
Since this is the main entry point (and probably the only function that needs to be public), give this a full description in comment.

Reply:
Done.